dataframe-jdbc/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/readJdbc.kt

```diff
@@ -0,0 +1,635 @@
+package org.jetbrains.kotlinx.dataframe.io
+
+import io.github.oshai.kotlinlogging.KotlinLogging
+import org.jetbrains.kotlinx.dataframe.AnyFrame
+import org.jetbrains.kotlinx.dataframe.DataFrame
+import org.jetbrains.kotlinx.dataframe.api.toDataFrame
+import org.jetbrains.kotlinx.dataframe.impl.schema.DataFrameSchemaImpl
+import org.jetbrains.kotlinx.dataframe.io.db.*
+import org.jetbrains.kotlinx.dataframe.schema.DataFrameSchema
+import java.sql.*
+import java.util.*
+
+private val logger = KotlinLogging.logger {}
+
+/**
+ * The default limit value.
+ *
+ * This constant represents the default limit value to be used in cases where no specific limit
+ * is provided.
+ *
+ * @see Int.MIN_VALUE
+ */
+private const val DEFAULT_LIMIT = Int.MIN_VALUE
+
+/**
+ * Represents a column in a database table to keep all required meta-information.
+ *
+ * @property [name] the name of the column.
+ * @property [sqlType] the SQL data type of the column.
+ * @property [jdbcType] the JDBC data type of the column.
+ * @property [size] the size of the column.
+ */
+public data class TableColumnMetadata(val name: String, val sqlType: String, val jdbcType: Int, val size: Int)
+
+/**
+ * Represents a table metadata to store information about a database table,
+ * including its name, schema name, and catalogue name.
+ *
+ * NOTE: we need to extract both, [schemaName] and [catalogue]
+ * because the different databases have different implementations of metadata.
+ *
+ * @property [name] the name of the table.
+ * @property [schemaName] the name of the schema the table belongs to (optional).
+ * @property [catalogue] the name of the catalogue the table belongs to (optional).
+ */
+public data class TableMetadata(val name: String, val schemaName: String?, val catalogue: String?)
+
+/**
+ * Represents the configuration for a database connection.
+ *
+ * @property [url] the URL of the database.
+ * @property [user] the username used for authentication (optional, default is empty string).
+ * @property [password] the password used for authentication (optional, default is empty string).
+ */
+public data class DatabaseConfiguration(val url:String, val user: String = "", val password: String = "")
```

Comment:
I tried this API without reading the docs (on purpose :) ), it wasn't clear for me what is URL. Only after some experiments i found what i should write something like this
"jdbc:sqlite:/home/path/chinook.db"

Reply:
Thanks, will improve and probably add one require about possible String formats